### PR TITLE
chore: introduce new library maturity badges

### PR DESCRIPTION
**Pull request by rix0rrr**
### Commit Message
chore: introduce new library maturity badges

Add a concept of library "maturity" in addition to "stability".

**Maturity** is a human concept that communicates how ready we
think the package is for public consumption, and is leading.

**Stability** is a mechanical concept that affects how `jsii-diff`
checks the package, and must follow `maturity`.

### End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: tools/pkglint/lib/rules.ts [hunk 1]
```diff
@@ -173,34 +173,69 @@ export class StabilitySetting extends ValidationRule {
...
+    // 'deprecated' or 'external'. This is the value that will be used for every class
+    // unless it is annotated otherwise. It must be one of these values, which will be
+    // enforced by 'jsii'.
+    //
+    // On top of this, we have conceptual library-level "maturity", which is one of
+    // the following stages: 'cfn-only', 'experimental', 'developer-preview', 'stable',
+    // 'deprecated'. These render into stability banners in the README.
+    //
+    // The maturities mostly map onto default stabilities, and so maturity will
+    // be derived from stability if possible. The mapping is:
```
Comment (eladb) [inline tools/pkglint/lib/rules.ts:186]: Feels like stability should derive from maturity, no?

Experimental, Developer preview => experimental
Stable => Stable

CFN classes should always be annotated with @external, no?

File: tools/pkglint/lib/rules.ts [hunk 2]
```diff
@@ -173,34 +173,69 @@ export class StabilitySetting extends ValidationRule {
...
+    //                    developer-preview
+    //     stable         stable
+    //     deprecated     deprecated
+    //     external       (not used, we don't allow 'external' as default stability)
+    //
+    // Since 'stability: experimental' maps onto 3 different maturities, you must annotate
+    // this manually. Otherwise, the stability value will be used to determine maturity.
+    //
+    // (Potentially, 'cfn-only' could have been derived automatically, but that would require
+    // making pkglint depend on 'jsii-reflect'... let's not go there).
```
Comment (eladb) [inline tools/pkglint/lib/rules.ts:200]: Why can't we just check if there are any files under `lib` besides `index.ts`?

File: tools/pkglint/lib/rules.ts (hunk 1 above)
Comment (rix0rrr) [reply tools/pkglint/lib/rules.ts:186]: Yeah, I guess that makes more sense. I was doing the minimal change from what we currently have. 

> CFN classes should always be annotated with @external, no?

And so they are. Still, "external" is a valid stability and hence can be put into the default "stability" field.

File: tools/pkglint/lib/rules.ts (hunk 2 above)
Comment (rix0rrr) [reply tools/pkglint/lib/rules.ts:200]: https://github.com/aws/aws-cdk/tree/master/packages/%40aws-cdk/aws-cloudtrail/lib

not so quick

Comment (rix0rrr) [reply tools/pkglint/lib/rules.ts:200]: Because we no consistency :)

And in any case, I think I'd prefer a model in which maturity is leading.

**Comment by rix0rrr** [comment]
> I think `package.json` is taken care of but the stability text seems out of sync.

OH FOR CRYING OUT LOUD

File: packages/@aws-cdk/aws-amplify/package.json [hunk 3]
```diff
@@ -109,6 +109,7 @@
     "node": ">= 10.12.0"
   },
   "stability": "experimental",
+  "maturity": "cfn-only",
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-amplify/package.json:112]: this module has some [constructs](https://docs.aws.amazon.com/cdk/api/latest/docs/aws-amplify-readme.html) now

File: packages/@aws-cdk/cx-api/README.md [hunk 4]
```diff
@@ -3,16 +3,14 @@
...
-
-> **This is a _developer preview_ (public beta) module.**
->
-> All classes with the `Cfn` prefix in this module ([CFN Resources](https://docs.aws.amazon.com/cdk/latest/guide/constructs.html#constructs_lib))
-> are auto-generated from CloudFormation. They are stable and safe to use.
->
-> However, all other classes, i.e., higher level constructs, are under active development and subject to non-backward
-> compatible changes or removal in any future version. These are not subject to the [Semantic Versioning](https://semver.org/) model.
-> This means that while you may use them, you may need to update your source code when upgrading to a newer version of this package.
+![cfn-resources: Stable](https://img.shields.io/badge/cfn--resources-stable-success.svg?style=for-the-badge)
```
Comment (shivlaks) [inline packages/@aws-cdk/cx-api/README.md:6]: does this make sense for this module? (there's a few others... I just pushed one to the `cloud-assembly-schema`.... doesn't feel right.

It's even confusing for `custom-resources` and any other documented module without a documented `cfn*` class. what do you think?

Comment (rix0rrr) [reply packages/@aws-cdk/cx-api/README.md:6]: Yeah, I had considered that as well, but didn't really want to make the logic even more complicated. But I can. Let me see.

**Comment by rix0rrr** [comment]
@shivlaks I believe I've addressed your concerns.

**Pull request merged**